gitfs/worker/sync.py

```diff
@@ -87,7 +87,9 @@ def on_idle(self):
                 self.commit(self.commits)
                 self.commits = []
             log.debug("Start syncing")
-            self.sync()
+            while not self.sync():
```

Comment:
The only thing to note here is that I believe that if the sync worker stalls new local commits won't be added to the queue for further pushing for that period.

But it shouldn't be the case. The max wait time is 33s.